src/pages/en/concepts/mpa-vs-spa.md

```diff
@@ -0,0 +1,90 @@
+
+Rendering your HTML in the browser may sound like the faster option vs. requesting it from a remote server. However, a true SPAs (with no server rendering) will be consistently slower than an MPA. This is because an SPA has to download, parse, and run an entire JavaScript application in the browser just to render any HTML on the page. Then, your SPA will likely need to fetch remote data anyway, introducing even more wait time before your page is finished loading.
+
+Most SPA frameworks will attempt to mitigate this performance problem by adding basic server-rendering on the first page load. Server rendering the first page gives your user something to look at while your site loads in the background. However, this can introduce a new "uncanny valley" problem where your site appears loaded but unresponsive, since the application logic is still loading in the background.
+
+MPAs render all HTML on a remote server and often don't require much JavaScript to run. This gives MPAs a much faster first load experience than SPAs, which is essential for content-focused websites. But this comes with a tradeoff: future page navigation can't benefit from local rendering, so long-lived user experiences may feel slower in MPAs vs. SPAs.
+
+
+#### Server routing (MPA) vs. client routing (SPA)
+
+Where does your website router live? In an MPA, every request to the server decides which HTML to respond with, so the routing logic lives in the server. In an SPA, your router runs right in the browser, and hijacks any navigation to render the new page locally without ever hitting a server.
+
+This is a similar tradeoff to the one described above: MPAs offer a faster first load experience, while SPAs sometimes ofer a faster second or third page load once the JavaScript application is fully loaded in the browser. 
+
+SPAs can also offer a more seamless navigation experience because they control rendering across the entire transition. To mitigate this, MPAs leverage tools like Hotwire's [Turbo](https://turbo.hotwired.dev/) that mimic client routing by also controlling navigation in the browser. The HTML is still rendered on the server, but Turbo can now display a seamless transition between pages similar to client routing in an SPA.
+
+#### Server state management (MPA) vs. client state management (SPA)
+
+SPAs are the superior model for complex state management. Because the entire website runs as a single JavaScript application in the browser, SPAs don't need to worry about maintaining state across page transitions. Both MPAs and most SPAs need to worry about server-side state for server rendering.
+
+SPAs really shine in complex websites that display lots of interactive data and need lots of JavaScript to run. Interactive, data-drive experiences like inboxes and admin dashboards do well as SPAs because the website itself is inherintely "app-like".
```

Comment:
```suggestion
SPAs really shine in complex websites that display lots of interactive data and need lots of JavaScript to run. Interactive, data-driven experiences like inboxes and admin dashboards do well as SPAs because the website itself is inherently "app-like".
```

Reply:
I keep coming here from reading the preview page to find that Yan has already found the typos!